operate/webapp/src/main/java/io/camunda/operate/webapp/elasticsearch/reader/OperationReader.java

```diff
@@ -313,9 +324,52 @@ public List<OperationDto> getOperationsByBatchOperationId(String batchOperationI
     }
   }
 
+  @Override
+  public List<BatchOperationDto> enrichBatchEntitiesWithMetadata(
```

Comment:
This should probably be in a separate component. The OperationReader class is intended for data fetching from search, this is more transformation of existing data the client already has. (This class is also already very large, it would be best to put this in its own transformer class, same with the Opensearch version)

Reply:
You're right. There are some parts in the code around here that don't quite follow the single responsibility principle, but I shouldn't be adding on top of that.
I'll change this tomorrow. Let me know if there's anything else you notice meanwhile.

Thank you!

Reply:
I moved the data transformation to an own component (DataAggregator), but kept the requests for reading Operations in the OperationReader as I thought that's the most fitting.
I also changed the requests to use ID-batches rather than request every ID individually and instead work more with aggregations.